diff --git a/node_modules/@typescript-eslint/parser/dist/parser.js b/node_modules/@typescript-eslint/parser/dist/parser.js
index 4b0453b..2d106e7 100644
--- a/node_modules/@typescript-eslint/parser/dist/parser.js
+++ b/node_modules/@typescript-eslint/parser/dist/parser.js
@@ -63,6 +63,8 @@ function parseForESLint(code, parserOptions) {
     else {
         parserOptions = { ...parserOptions };
     }
+    console.log(code);
+    console.log(parserOptions);
     // https://eslint.org/docs/user-guide/configuring#specifying-parser-options
     // if sourceType is not provided by default eslint expect that it will be set to "script"
     if (parserOptions.sourceType !== 'module' &&
@@ -99,6 +101,7 @@ function parseForESLint(code, parserOptions) {
         sourceType: parserOptions.sourceType,
     };
     const { ast, services } = (0, typescript_estree_1.parseAndGenerateServices)(code, tsestreeOptions);
+    console.log(ast);
     ast.sourceType = parserOptions.sourceType;
     if (services.program) {
         // automatically apply the options configured for the program
