diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js
index a2d4898..3cd7a65 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js
@@ -15,6 +15,7 @@ const log = (0, debug_1.default)('typescript-eslint:typescript-estree:createProj
  */
 function createProjectProgram(parseSettings, programsForProjects) {
     log('Creating project program for: %s', parseSettings.filePath);
+    console.log(programsForProjects);
     const astAndProgram = (0, node_utils_1.firstDefined)(programsForProjects, currentProgram => (0, shared_1.getAstFromProgram)(currentProgram, parseSettings.filePath));
     if (!astAndProgram) {
         throw new Error((0, createProjectProgramError_1.createProjectProgramError)(parseSettings, programsForProjects).join('\n'));
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js
index 8bb3abb..4940e65 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js
@@ -213,6 +213,7 @@ function createWatchProgram(tsconfigPath, parseSettings) {
     const oldReadFile = watchCompilerHost.readFile;
     watchCompilerHost.readFile = (filePathIn, encoding) => {
         const filePath = (0, shared_1.getCanonicalFileName)(filePathIn);
+        console.log(filePath, currentLintOperationState.filePath);
         const fileContent = filePath === currentLintOperationState.filePath
             ? (0, source_files_1.getCodeText)(currentLintOperationState.code)
             : oldReadFile(filePath, encoding);
