diff --git a/node_modules/eslint/lib/linter/linter.js b/node_modules/eslint/lib/linter/linter.js
index db059b2..76b7a6b 100644
--- a/node_modules/eslint/lib/linter/linter.js
+++ b/node_modules/eslint/lib/linter/linter.js
@@ -2074,7 +2074,7 @@ class Linter {
 						loc,
 					} of inlineConfigResult.configs) {
 						Object.keys(inlineConfig.rules).forEach(ruleId => {
-							const rule = config.getRuleDefinition(ruleId);
+              const rule = (providedOptions.getRule ?? config.getRuleDefinition)(ruleId);
 							const ruleValue = inlineConfig.rules[ruleId];
 
 							if (!rule) {
@@ -2233,11 +2233,18 @@ class Linter {
 			options.allowInlineConfig && !options.warnInlineConfig
 				? getDirectiveCommentsForFlatConfig(
 						sourceCode,
-						ruleId => config.getRuleDefinition(ruleId),
+						ruleId => (providedOptions.getRule ?? config.getRuleDefinition)(ruleId),
 						config.language,
 					)
 				: { problems: [], disableDirectives: [] };
     
+    if (typeof providedOptions.patchInlineOptions === "function") {
+      providedOptions.patchInlineOptions(mergedInlineConfig);
+    }
+    if (typeof providedOptions.patchDirectives === "function") {
+      providedOptions.patchDirectives(commentDirectives);
+    }
+
 		const configuredRules = Object.assign(
 			{},
 			config.rules,
