diff --git a/node_modules/eslint/lib/linter/linter.js b/node_modules/eslint/lib/linter/linter.js
index 1baa0b8..4a2347c 100644
--- a/node_modules/eslint/lib/linter/linter.js
+++ b/node_modules/eslint/lib/linter/linter.js
@@ -1826,7 +1826,7 @@ class Linter {
                     for (const { config: inlineConfig, loc } of inlineConfigResult.configs) {
 
                         Object.keys(inlineConfig.rules).forEach(ruleId => {
-                            const rule = getRuleFromConfig(ruleId, config);
+                            const rule = (providedOptions.getRule ?? getRuleFromConfig)(ruleId, config);
                             const ruleValue = inlineConfig.rules[ruleId];
 
                             if (!rule) {
@@ -1965,11 +1965,19 @@ class Linter {
         const commentDirectives = options.allowInlineConfig && !options.warnInlineConfig
             ? getDirectiveCommentsForFlatConfig(
                 sourceCode,
-                ruleId => getRuleFromConfig(ruleId, config),
+                ruleId => (providedOptions.getRule ?? getRuleFromConfig)(ruleId, config),
                 config.language
             )
             : { problems: [], disableDirectives: [] };
 
+        if (typeof providedOptions.patchInlineOptions === "function") {
+          providedOptions.patchInlineOptions(mergedInlineConfig);
+        }
+
+        if (typeof providedOptions.patchDirectives === "function") {
+          providedOptions.patchDirectives(commentDirectives);
+        }
+
         const configuredRules = Object.assign({}, config.rules, mergedInlineConfig.rules);
 
         let lintingProblems;
