diff --git a/node_modules/eslint/lib/config/rule-validator.js b/node_modules/eslint/lib/config/rule-validator.js
index 3b4ea61..6eb854f 100755
--- a/node_modules/eslint/lib/config/rule-validator.js
+++ b/node_modules/eslint/lib/config/rule-validator.js
@@ -167,6 +167,7 @@ class RuleValidator {
                 validateRule(ruleOptions.slice(1));
 
                 if (validateRule.errors) {
+                    console.log(JSON.stringify(validateRule.errors));
                     throw new Error(`Key "rules": Key "${ruleId}":\n${
                         validateRule.errors.map(
                             error => {
