diff --git a/node_modules/eslint/lib/linter/linter.js b/node_modules/eslint/lib/linter/linter.js
index f46f5e3..aa203b9 100755
--- a/node_modules/eslint/lib/linter/linter.js
+++ b/node_modules/eslint/lib/linter/linter.js
@@ -1934,7 +1934,7 @@ class Linter {
 						loc,
 					} of inlineConfigResult.configs) {
 						Object.keys(inlineConfig.rules).forEach(ruleId => {
-							const rule = config.getRuleDefinition(ruleId);
+              const rule = (providedOptions.getRule ?? config.getRuleDefinition)(ruleId);
 							const ruleValue = inlineConfig.rules[ruleId];
 
 							if (!rule) {
@@ -2084,12 +2084,19 @@ class Linter {
 			options.allowInlineConfig && !options.warnInlineConfig
 				? getDirectiveCommentsForFlatConfig(
 						sourceCode,
-						ruleId => config.getRuleDefinition(ruleId),
+            ruleId => (providedOptions.getRule ?? config.getRuleDefinition)(ruleId),
 						config.language,
 						report,
 					)
 				: [];
 
+    if (typeof providedOptions.patchInlineOptions === "function") {
+      providedOptions.patchInlineOptions(mergedInlineConfig);
+    }
+    if (typeof providedOptions.patchDirectives === "function") {
+      providedOptions.patchDirectives(commentDirectives);
+    }
+
 		const configuredRules = Object.assign(
 			{},
 			config.rules,
