<p>There’s no reason to import modules you don’t use; and every reason not to: doing so needlessly increases the load.</p>
<h2>Noncompliant Code Example</h2>
<pre>
import A from 'a';      // Noncompliant, A isn't used
import { B1 } from 'b';

console.log(B1);
</pre>
<h2>Compliant Solution</h2>
<pre>
import { B1 } from 'b';

console.log(B1);
</pre>

