<p>Doing an operation on a string without using the result of the operation is useless and is certainly due to a misunderstanding.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var str = "..."
str.toUpperCase(); // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
var str = "..."
str = str.toUpperCase();
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated; use {rule:javascript:S2201} instead.</p>

