A <code>for</code> loop with a stop condition that can never be reached, such as one with a counter that moves in the wrong direction, will run infinitely. While there are occasions when an infinite loop is intended, the convention is to construct such loops as <code>while</code> loops. More typically, an infinite <code>for</code> loop is a bug. 

<h2>Noncompliant Code Example</h2>
<pre>
for (var i = 0; i < strings.length; i--) { // Noncompliant;
  //...
}
</pre>

<h2>Compliant Solution</h2>
<pre>
for (var i = 0; i < strings.length; i++) {
  //...
}
</pre>