<p>
  The <code>parseInt</code> function has two versions, one that takes a base value as a second argument and one that does not, and using the single-arg version can result in unexpected results on older browsers.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
parseInt("010");  // Noncompliant; pre-2013 browsers may return 8
</pre>

<h2>Compliant Solution</h2>
<pre>
parseInt("010", 10);
</pre>
