<p>A test case without assertions ensures only that no exceptions are thrown. Beyond basic runnability, it ensures nothing about the behavior of the
code under test.</p>
<p>This rule raises an exception when the assertion library <code>chai</code> is imported but no assertion is used in a test.</p>
<h2>Noncompliant Code Example</h2>
<pre>
const expect = require('chai').expect;

describe("No assertion", function() {
    it("doesn't test anything", function() { // Noncompliant
        const str = "";
    });
});
</pre>
<h2>Compliant Solution</h2>
<pre>
const expect = require('chai').expect;

describe("Has assertions", function() {
    it("tests a string", function() {
        const str = "";
        expect(str).to.be.a('string');
    });
});
</pre>

