<p>The <code>void</code> operator evaluates its argument and unconditionally returns <code>undefined</code>. It can be useful in pre-ECMAScript 5
environments, where <code>undefined</code> could be reassigned, but generally, its use makes code harder to understand.</p>
<h2>Noncompliant Code Example</h2>
<pre>
void doSomething();
</pre>
<h2>Compliant Solution</h2>
<pre>
doSomething();
</pre>
<h2>Exceptions</h2>
<p>No issue is raised when <code>void 0</code> is used in place of <code>undefined</code>.</p>
<pre>
if (parameter === void 0) {...}
</pre>
<p>No issue is raised when <code>void</code> is used before immediately invoked function expressions.</p>
<pre>
void (function() {
   ...
}());
</pre>
<p>No issue is raised when <code>void</code>'s argument is a promise.</p>
<pre>
const promise = new Promise((resolve, reject) =&gt; {
   setTimeout(() =&gt; {
      resolve('done');
   }, 3000);
});
void promise;
</pre>

