<p>An <code>async</code> function always wraps the return value in a <code>Promise</code>. Using <code>return await</code> is not required and comes
at an extra performance cost. However, you might wish to keep it as it preserves the function call in the stack trace in case an error is thrown
asynchronously.</p>
<h2>Noncompliant Code Example</h2>
<pre>
async function foo() {
  // ...
}

async function bar() {
  // ...
  return await foo(); // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
async function foo() {
  // ...
}

async function bar() {
  // ...
  return foo();
}
</pre>

