<p>React expects a unique identifier for performance optimizations. Using generated values like <code>Math.random()</code> or <code>Date.now()</code>
is discouraged as it would force a re-render every time and might also provoke bugs if values collide.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function generateButtons(props) {
  return props.buttons.map((button, index) =&gt; {
    &lt;Button key={Math.random()}&gt;button.number&lt;/Button&gt;
  });
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function generateButtons(props) {
  return props.buttons.map((button, index) =&gt; {
    &lt;Button key={button.number}&gt;button.number&lt;/Button&gt;
  });
}
</pre>
<h2>See</h2>
<ul>
  <li> <a
  href="https://reactjs.org/docs/reconciliation.html#recursing-on-children">https://reactjs.org/docs/reconciliation.html#recursing-on-children</a> -
  React API reference </li>
  <li> {rule:javascript:S6477} </li>
  <li> {rule:javascript:S6479} </li>
</ul>

