<p>
  In JavaScript, the semicolon is optional as a statement separator, but omitting semicolons can be confusing.
</p>


<h2>Noncompliant Code Example</h2>
<p>
  Here is an example of such confusing code:
</p>
<pre>
function fun() {
  return
       5
}
print(fun());
</pre>

<p>
  Will print "undefined", because interpreted as:
</p>
<pre>
function fun() {
  return;
  5;
}
print(fun());
</pre>

<h2>Compliant Solution</h2>
<pre>
function fun() {
  return 5;
}
print(fun());
</pre>
