<p>
  For consistency single-quotes (') are preferred to double-quotes (").
  This is helpful when creating strings that include HTML.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
var firstParameter = "something";
</pre>

<h2>Compliant Solution</h2>
<pre>
var firstParameter = 'something';
</pre>
