<h2>Why is this an issue?</h2>
<p>An invalid color definition will by default be interpreted as black, which is likely to have unintended impacts on the expected look and feel of
the website.</p>
<p>This rule raises an issue when a color definition (<code>color</code>, <code>background-color</code>) is not valid. The color definition is
considered valid when it is made of hexadecimal characters:</p>
<ul>
  <li> longhand: 6 or 8 characters (when alpha is defined) </li>
  <li> shorthand variant: 3 or 4 characters (when alpha is defined) </li>
</ul>
<h3>Noncompliant code example</h3>
<pre>
a {
 color: #3c; /* Noncompliant; shorthand should be made of 3 characters */
}
div {
  background-color: #3cb371a; /* Noncompliant; alpha should have 2 characters */
}
</pre>
<h3>Compliant solution</h3>
<pre>
a {
 color: #3cc;
}
div {
  background-color: #3cb371ac;
}
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/color_value">Mozilla Web Technology for Developers</a> - CSS Color </li>
</ul>

