<h2>Why is this an issue?</h2>
<p>Having duplicated font names doesnâ€™t help to read the font declaration and may be an indicator the author of the line was not sure how to configure
it. This rule raises an issue when <code>font</code> or <code>font-family</code> properties contain a duplicated font name. This rule ignores
<code>$sass</code>, <code>@less</code>, and <code>var(--custom-property)</code> variable syntaxes.</p>
<h3>Noncompliant code example</h3>
<pre>
a {
  font-family: 'Georgia', Georgia, serif; /* Noncompliant; 'Georgia' is duplicated */
}
</pre>
<h3>Compliant solution</h3>
<pre>
a {
  font-family: Georgia, serif;
}
</pre>

