<h2>Why is this an issue?</h2>
<p>The W3C specifications define the <code>units</code> that can be used with lengths. A unit that is not part of the list of supported ones is likely
to be a typo and will be seen as a UI bug by the user.</p>
<p>This rule raises an issue each time a unit is not officially supported.</p>
<h3>Noncompliant code example</h3>
<pre>
a {
  width: 10pixels; /* Noncompliant; "pixels" is not a valid unit */
}
</pre>
<h3>Compliant solution</h3>
<pre>
a {
  width: 10px;
}
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://www.w3.org/TR/css3-values/#lengths">CSS Specification</a> - Distance units </li>
</ul>

