<h2>Why is this an issue?</h2>
<p>A shorthand property defined after a longhand property will completely override the value defined in the longhand property making the longhand one
useless. The code should be refactored to consider the longhand property or to remove it completely.</p>
<h3>Noncompliant code example</h3>
<pre>
a {
  padding-left: 10px;
  padding: 20px; /* Noncompliant; padding is overriding padding-left making it useless */
}
</pre>
<h3>Compliant solution</h3>
<pre>
a {
  padding: 10px; /* Compliant; padding is defining a general behaviour and padding-left, just after, is precising the left case */
  padding-left: 20px;
}
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties">Mozilla Web Technology for Developers</a> - Shorthand
  properties </li>
</ul>

