<h2>Why is this an issue?</h2>
<p>CSS allows duplicate property names but only the last instance of a duplicated name determines the actual value that will be used for it.
Therefore, changing values of other occurrences of a duplicated name will have no effect and may cause misunderstandings and bugs.</p>
<p>This rule ignores <code>$sass</code>, <code>@less</code>, and <code>var(--custom-property)</code> variable syntaxes.</p>
<h3>Noncompliant code example</h3>
<pre>
a {
  color: pink;
  background: orange;
  color: orange
}
</pre>
<h3>Compliant solution</h3>
<pre>
a {
  color: pink;
  background: orange
}
</pre>

