<h2>Why is this an issue?</h2>
<p>The W3C specifications define the valid pseudo-class selectors. Only the official and browser-specific pseudo-class selectors should be used to get
the expected impact in the final rendering.</p>
<h3>Noncompliant code example</h3>
<pre>
a:hoverr { /* Noncompliant; there is a typo on the word "hover" */
...
}
</pre>
<h3>Compliant solution</h3>
<pre>
a:hover {
...
}
</pre>

