<h2>Why is this an issue?</h2>
<p>Leftover empty blocks are usually introduced by mistake. They are useless and prevent readability of the code. They should be removed or completed
with real code.</p>
<h3>Noncompliant code example</h3>
<pre>
a { }
</pre>
<h3>Compliant solution</h3>
<pre>
a { color: pink; }
</pre>

