<h2>Why is this an issue?</h2>
<p>The W3C specifications define the valid CSS properties. Only the official and browser-specific properties should be used to get the expected impact
in the final rendering.</p>
<p>This rule ignores:</p>
<ul>
  <li> <code>$sass</code>, <code>@less</code>, and <code>var(--custom-property)</code> variable syntaxes. </li>
  <li> vendor-prefixed properties (e.g., <code>-moz-align-self</code>, <code>-webkit-align-self</code>). </li>
</ul>
<h3>Noncompliant code example</h3>
<pre>
a {
  colour: blue; /* Noncompliant; colour is not part of the specifications */
}
</pre>
<h3>Compliant solution</h3>
<pre>
a {
  color: blue;
}
</pre>

