<h2>Why is this an issue?</h2>
<p>The W3C specifications define the valid <code>at-rules</code>. Only the official and browser-specific <code>at-rules</code> should be used to get
the expected impact in the final rendering.</p>
<h3>Noncompliant code example</h3>
<pre>
@encoding "utf-8";
</pre>
<h3>Compliant solution</h3>
<pre>
@charset "utf-8";
</pre>

