<h2>Why is this an issue?</h2>
<p>The W3C specifications define the valid pseudo-element selectors. Only the official and browser-specific pseudo-element selectors should be used to
get the expected impact in the final rendering.</p>
<h3>Noncompliant code example</h3>
<pre>
a::beforre { /* Noncompliant; there is a typo on the word "before" */
...
}
</pre>
<h3>Compliant solution</h3>
<pre>
a::before {
...
}
</pre>

