<h2>Why is this an issue?</h2>
<p>HTML, SVG, and MathML define the selectors which can be used in a CSS. A selector that is not part of them is likely to be a typo or a
misunderstanding of the CSS syntax.</p>
<h3>Noncompliant code example</h3>
<pre>
field {}

ul list {}
</pre>
<h3>Compliant solution</h3>
<pre>
input {}

ul li {}
</pre>

