<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>To perform calculations when specifying a CSS property <code>calc()</code> function can be used. This function takes single expression as
parameter. When writing this expression some rules must be respected:</p>
<ul>
  <li> no empty <code>calc()</code> </li>
  <li> there should be an operator between the arguments </li>
  <li> there should not be any division by zero </li>
</ul>
<p>Otherwise <code>calc()</code> function will be invalid and the entire rule using it will be ignored.</p>
<h3>Noncompliant code example</h3>
<pre>
.btn {
  border: solid black 1px;
  width: calc(100% 80px);  /* Noncompliant */
}
</pre>
<h3>Compliant solution</h3>
<pre>
.btn {
  border: solid black 1px;
  width: calc(100% - 80px);
}
</pre>

