<h2>Why is this an issue?</h2>
<p>A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many
things.</p>
<h3>Noncompliant code example</h3>
<p>With a maximum number of 4 parameters:</p>
<pre>
function doSomething(param1, param2, param3, param4, param5) {
...
}
</pre>
<h3>Compliant solution</h3>
<pre>
function doSomething(param1, param2, param3, param4) {
...
}
</pre>
<h3>Exceptions</h3>
<p>The rule ignores constructors where parameters are <strong>all</strong> parameter properties:</p>
<pre>
class C {
  constructor(
    private param1: number,
    private param2: boolean,
    private param3: string,
    private param4: string[],
    private param5: number | string
  ) {}
}
</pre>
<p>The rule also ignores Angular component constructors:</p>
<pre>
import { Component } from '@angular/core';

@Component({/* ... */})
class Component {
  constructor(p1, p2, p3, p4, p5) {}
}
</pre>

