<p>This rule is deprecated; use {rule:javascript:S2201} instead.</p>
<h2>Why is this an issue?</h2>
<p>Doing an operation on a string without using the result of the operation is useless and is certainly due to a misunderstanding.</p>
<h3>Noncompliant code example</h3>
<pre>
var str = "..."
str.toUpperCase(); // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
var str = "..."
str = str.toUpperCase();
</pre>

