<h2>Why is this an issue?</h2>
<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule checks that all variable,
property and parameter names match a provided regular expression.</p>
<h3>Noncompliant code example</h3>
<p>With the default regular expression <code>^[_$A-Za-z][$A-Za-z0-9]*$|^[_$A-Z][_$A-Z0-9]+$</code>:</p>
<pre>
const foo_bar = 1;
const baz_ = 2;
</pre>
<h3>Compliant solution</h3>
<pre>
const fooBar = 1;
const _baz = 2;
</pre>

