<h2>Why is this an issue?</h2>
<p>This rule checks that all string literals use the same kind of quotes.</p>
<h3>Noncompliant code example</h3>
<p>Using the parameter default (forcing single quotes):</p>
<pre>
var firstParameter = "something"; // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
var firstParameter = 'something';
</pre>
<h3>Exceptions</h3>
<p>Strings that contain quotes are ignored.</p>
<pre>
let heSaid = "Then he said 'What?'."  // ignored
let sheSaid = '"Whatever!" she replied.'  // ignored
</pre>
