<h2>Why is this an issue?</h2>
<p>Most browsers parse and discard a meaningless, trailing comma. Unfortunately, thatâ€™s not the case for Internet Explorer below version 9, which
throws a meaningless error. Therefore trailing commas should be eliminated.</p>
<h3>Noncompliant code example</h3>
<pre>
var settings = {
    'foo'  : oof,
    'bar' : rab,    // Noncompliant - trailing comma
};
</pre>
<h3>Compliant solution</h3>
<pre>
var settings = {
    'foo'  : oof,
    'bar' : rab
};
</pre>
