<p>While most script engines support function declarations within blocks, from browser to browser, the implementations are inconsistent with each
other.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (x) {
  function foo() {} //foo is hoisted in Chrome, Firefox and Safari, but not in Edge.
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if (x) {
  const foo = function() {}
}
</pre>

