<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>Most browsers parse and discard a meaningless, trailing comma. Unfortunately, thatâ€™s not the case for Internet Explorer below version 9, which
throws a meaningless error. Therefore trailing commas should be eliminated.</p>
<h3>Noncompliant code example</h3>
<pre>
var settings = {
    'foo'  : oof,
    'bar' : rab,    // Noncompliant - trailing comma
};
</pre>
<h3>Compliant solution</h3>
<pre>
var settings = {
    'foo'  : oof,
    'bar' : rab
};
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> <a href="https://eslint.style/">ESLint Stylistic</a> - Rule <a href="https://eslint.style/rules/comma-dangle">comma-dangle</a> </li>
</ul>
