<h2>Why is this an issue?</h2>
<p>While most script engines support function declarations within blocks, from browser to browser, the implementations are inconsistent with each
other.</p>
<h3>Noncompliant code example</h3>
<pre>
if (x) {
  function foo() {} //foo is hoisted in Chrome, Firefox and Safari, but not in Edge.
}
</pre>
<h3>Compliant solution</h3>
<pre>
if (x) {
  const foo = function() {}
}
</pre>

