<h2>Why is this an issue?</h2>
<p>There is no good reason to create a new object to not do anything with it. Most of the time, this is due to a missing piece of code and so could
lead to an unexpected behavior in production.</p>
<p>If it was done on purpose because the constructor has side-effects, then that side-effect code should be moved into a separate method and called
directly.</p>
<h3>Noncompliant code example</h3>
<pre>
new MyConstructor(); // Non-Compliant
</pre>
<h3>Compliant solution</h3>
<pre>
var something = new MyConstructor();  // Compliant
</pre>
<h3>Exceptions</h3>
<p>Immediately dropped new objects inside <code>try</code>-statements are ignored.</p>
<pre>
try {
  new MyConstructor();
} catch (e) {
  /* ... */
}
</pre>

