<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>Modern browsers ignore unneeded, trailing commas, so there are no negatives to having them unless youâ€™re supporting an IE 8 application. Since they
make adding new properties simpler, their use is preferred. This rule raises an issue when the last item in a multiline construct (array or object
literal, import or export statement, function declaration or call) does not end with a trailing comma and does not lie on the same line as the closing
curly brace, bracket or parenthesis.</p>
<h3>Noncompliant code example</h3>
<pre>
var joe = {
  fname: "Joe",
  lname: "Smith"      // Noncompliant
};
</pre>
<h3>Compliant solution</h3>
<pre>
var joe = {
  fname: "Joe",
  lname: "Smith",    // OK
};

var joe = {
  fname: "Joe",
  lname: "Smith"};   // OK
</pre>

