<h2>Why is this an issue?</h2>
<p>Template strings allow developers to embed variables or expressions in strings using template literals, instead of string concatenation. This is
done by using expressions like <code>${variable} </code> in a string between two back-ticks (<code>`</code>). However, when used in a regular string
literal (between double or single quotes) the template will not be evaluated and will be used as a literal, which is probably not what was
intended.</p>
<h3>Noncompliant code example</h3>
<pre>
console.log("Today is ${date}"); // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
console.log(`Today is ${date}`);
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> <a href="https://eslint.org">eslint</a> - Rule <a
  href="https://eslint.org/docs/latest/rules/no-template-curly-in-string">no-template-curly-in-string</a> </li>
</ul>
