<h2>Why is this an issue?</h2>
<p>Type assertion can be done in two ways: with <code>as MyType</code> or with <code>&lt;MyType&gt;</code>. But since there is an ambiguity in the
latter when using JSX and there is no ambiguity in the former, <code>as</code> is preferred.</p>
<h3>Noncompliant code example</h3>
<pre>
var foo = &lt;any&gt;"foo";  // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
var foo = "foo" as any;
</pre>
