<h2>Why is this an issue?</h2>
<p>If a <code>&lt;td&gt;</code> cell has a <code>headers</code> attribute, it should reference only IDs of headers in the same column and row.</p>
<p>Note that it is usually better to use <code>scope</code> attributes of <code>&lt;th&gt;</code> headers instead of <code>headers</code> attribute.
<code>headers</code> attribute requires you to list every corresponding <code>&lt;th&gt;</code> header’s <code>id</code>, which is error-prone and
makes the code less maintainable. See&nbsp;<a href="https://www.w3.org/WAI/tutorials/tables/tips/">W3C WAI&nbsp;Web Accessibility
Tutorials</a>&nbsp;for more information.</p>
<p>If your table is too complex, it might be better to split it into multiple small tables as it improves both readability and maintainability.</p>
<p>This rule raises an issue when the <code>headers</code> attribute of a <code>&lt;td&gt;</code> cell contains IDs which don’t belong to a header in
the same row or column.</p>
<h3>Noncompliant code example</h3>
<pre>
&lt;table border="1"&gt;
&lt;caption&gt;
    Rental price
&lt;/caption&gt;
&lt;thead&gt;
    &lt;tr&gt;
        &lt;td&gt;&lt;/td&gt;
        &lt;th id="small" scope="col"&gt;
            Small car
        &lt;/th&gt;
        &lt;th id="big" scope="col"&gt;
            Big Car
        &lt;/th&gt;
    &lt;/tr&gt;
&lt;/thead&gt;
&lt;tbody&gt;
    &lt;tr&gt;
        &lt;th id="paris" class="span" colspan="3" scope="colgroup"&gt;
            Paris
        &lt;/th&gt;
        &lt;/tr&gt;
        &lt;tr&gt;
        &lt;th headers="paris" id="day1"&gt;
            1 day
        &lt;/th&gt;
        &lt;td headers="paris day1 big"&gt; &lt;!-- Noncompliant, referencing the column "big" instead of "small" --&gt;
            11 euros
        &lt;/td&gt;
        &lt;td headers="berlin day1 big"&gt; &lt;!-- Noncompliant, there is no header with id "berlin" --&gt;
            50 euros
        &lt;/td&gt;
        &lt;/tr&gt;
    &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-content-structure-separation-programmatic">WCAG2, 1.3.1</a>&nbsp;-&nbsp;Info
  and Relationships </li>
  <li> <a href="https://www.w3.org/TR/WCAG20-TECHS/html.html#H43">WCAG2, H43</a> - Using id and headers attributes to associate data cells with header
  cells in data tables </li>
</ul>

