<p>Testing frameworks like Mocha and Jest provide a way to run a single test case for a file by appending <code>.only()</code> to the test function.
This is convenient when debugging a specific use case. It makes, however, no sense to have it in production or development, and it was likely left by
mistake after such a debugging session. Therefore, it is strongly recommended not to commit usages of <code>.only()</code> to version control.</p>
<h2>Noncompliant Code Example</h2>
<pre>
describe("MyClass", function () {
    it.only("should run correctly", function () {
        /*...*/
    });
});
</pre>
<h2>Compliant Solution</h2>
<pre>
describe("MyClass", function () {
    it("should run correctly", function () {
        /*...*/
    });
});
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://mochajs.org/#exclusive-tests">exclusive tests in Mocha</a> </li>
  <li> <a href="https://jestjs.io/docs/next/api#testonlyname-fn-timeout">test.only() in Jest</a> </li>
  <li> <a href="https://jestjs.io/docs/next/api#describeonlyname-fn">describe.only() in Jest</a> </li>
</ul>

