<h2>Why is this an issue?</h2>
<p>TypeScript allows assigning values to enum members. However, it does not enforce uniqueness between the values of such members. As a result,
assigning the same value to different members is possible.</p>
<p>The rule raises issues on duplicate values because they are misleading and can lead to bugs that are hard to track down.</p>
<h2>How to fix it</h2>
<p>Assign a unique value for each enum member or donâ€™t assign any values.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
enum E {
  X = 0,
  Y = 0,
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
enum E {
  X = 0,
  Y = 1,
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> <a href="https://typescript-eslint.io/">typescript-eslint</a> - Rule <a
  href="https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-duplicate-enum-values.mdx">no-duplicate-enum-values</a> </li>
  <li> TypeScript Documentation - <a href="https://www.typescriptlang.org/docs/handbook/enums.html">Enums</a> </li>
</ul>
